megalista_dataflow/uploaders/google_analytics/google_analytics_measurement_protocol.py

```diff
@@ -38,29 +38,62 @@ def start_bundle(self):
   def _format_hit(self, payload: Dict[str, Any]) -> str:
     return "&".join([key + "=" + quote(str(value)) for key, value in payload.items() if value is not None])
 
-  @utils.safe_process(logger=logging.getLogger("megalista.GoogleAnalyticsMeasurementProtocolUploader"))
-  def process(self, batch: Batch, **kwargs):
-    execution = batch.execution
-    rows = batch.elements
+  def build_hit(self, batch: Batch, row: Dict[str, Any]) -> Dict[str, Any]:
+    metadata_list = batch.execution.destination.destination_metadata
 
-    # parameters starting with ** are optional.
-    payloads = [{
+    # initialize payload with values that are common to all types of hit
+    payload = {
       "v": 1,
-      "tid": execution.destination.destination_metadata[0],
-      "ni": execution.destination.destination_metadata[1],
-      "t": "event",
```

Comment:
I think we could move the line 59 above the line 45, setting the property"t" with hit_type and eliminating the lines 62, 68, 76.

Reply:
Nice catch, thanks!